docs/HowToDebug.md

```diff
@@ -1,4 +1,80 @@
 **How to Debug in NNI**
 ===
 
-*Coming soon*
+## Overview
+
+There are three parts that might have logs in NNI. They are nnimanager, dispatcher and trial. Here we will introduce them succinctly. More information please refer to [Overview](Overview.md).
+
+- **nnictl**: NNI controller is the nni command line tool that is used to started NNI.
+- **nnimanager**: nnimanager is the core of NNI, whose log is important when the whole experiment fails (e.g., no webUI or training service fails)
+- **Dispatcher**: Dispatcher is the collective name of **Tuner** and **Assessor**. Logs of dispatcher are related to the tuner or assessor code.
+    - **Tuner**: Tuner is an AutoML algorithm, which generates a new configuration for the next try. A new trial will run with this configuration.
+    - **Assessor**: Assessor analyzes trial's intermediate results (e.g., periodically evaluated accuracy on test dataset) to tell whether this trial can be early stopped or not.
+- **Trial**: Trial code is the code you write to run your experiment, which is an individual attempt at applying a new configuration (e.g., a set of hyperparameter values, a specific nerual architecture).
+
+## Where is the log
+
+### nnictl
+
+All possible errors that happen when launching NNI can be found here.
+
+You can use `nnictl log stderr` to find error information. Or find it in `~/nni/nnictl/log/stderr`.
+
+
+### Experiment Root Directory
+Every experiment has a root folder, which is shown on the right-top corner of webUI. Or you could assemble it by replacing the `experiment_id` with your actual experiment_id in path `~/nni/experiment/experiment_id/` in case of webUI failure. `experiment_id` could be seen when you run `nnictl create ...` to create a new experiment.
+
+Under the root path, there is a directory named `log`, where `nnimanager.log` and `dispatcher.log` are there.
+
+### Trial Root Directory
+
+Usually in webUI, you can click `+` in the left of every trial to expand it to see each trial's log path. Also there is another directory under experiment root directory, named `trials-local`, which stores all the trials run. Every trial has a unique id that is the name of its root directory. In this directory, a file named `stderr` records trial error and another named `trial.log` records this trial's log. 

```

Comment:
`trials-local`?

Reply:
`trials run` -> `trials`
`that is the name of its root directory` -> `as its directory name`